### PR TITLE
fix(form): add reset event and datepicker validation information

### LINKED ISSUE #197: bal-select has no fallback to react for form reset
Issue from => https://github.com/baloise/design-system/issues/641
Openedby => @Sw0rdiX

## Detailed description

> Describe in detail the issue you're having.
- If u reset a Form, the default value for bal-select will not be selected anymore...

> What did you expect to happen? What happened instead? What would you like to see changed?
- If i reset a Form, the value which is put in the bal-select as selected value will be selected again. 

> What browser are you working in?
- Chrome

## Steps to reproduce the issue

image

1. select another value as selected
2. reset form 

## Additional information

- Screenshots of Working Code
image

### LINKED ISSUE #364: [bal-datepicker] ValidityState for rangeOverflow/rangeUnderflow missing
## Detailed description

Since the bal-datepicker is using a plain input with type text to show/send the value of the custom-datepicker, it doesn't send the "real" ValidityStates" a type=datepicker input would send, e.g. rangeOverflow/rangeUnderflow

## Steps to reproduce the issue

1. Use bal-datepicker with a limitation
2. Manually set the text-input to something out of the range limits
3. Check the inputs ValidityState via JS
4. valid, no overflow/underflow.

Should be:
4. invalid - ValidityState rangeUnderflow/rangeOverflow

### PR DESCRIPTION
**Pull request by hirsch88**
Closes #197, #364

### CODE REVIEW

File: packages/components/src/components/form/bal-input/bal-input.tsx [hunk 1]
```diff
@@ -316,6 +325,7 @@ export class Input implements ComponentInterface, FormInput<string | undefined>
   }
 
   private onInput = (ev: InputEvent) => {
+    console.log('onInput')
```
Comment (nobilo) [inline packages/components/src/components/form/bal-input/bal-input.tsx]: this sneaky thingy

File: packages/components/src/styles/global/_all.scss [hunk 2]
```diff
@@ -1,3 +1,17 @@
...
+  left: 0;
+  background: transparent;
+  color: transparent;
+  border: none;
+  box-shadow: none;
+}
+
+input.is-one-pixel,
+select.is-one-pixel {
+  top: 32px;
```
Comment (nobilo) [inline packages/components/src/styles/global/_all.scss]: rem?

Comment (hirsch88) [reply packages/components/src/styles/global/_all.scss]: thanks done

File: packages/components/src/components/form/bal-input/bal-input.tsx (hunk 1 above)
Comment (hirsch88) [reply packages/components/src/components/form/bal-input/bal-input.tsx]: 😆

**Review by nobilo — APPROVED** [review]

**Pull request merged**